### PR TITLE
feat: add batches API with OpenAI compatibility (with inference replay)

### PR DESCRIPTION
**Pull request by mattf**
Add complete batches API implementation with protocol, providers, and tests:

Core Infrastructure:
- Add batches API protocol using OpenAI Batch types directly
- Add Api.batches enum value and protocol mapping in resolver
- Add OpenAI "batch" file purpose support
- Include proper error handling (ConflictError, ResourceNotFoundError)

Reference Provider:
- Add ReferenceBatchesImpl with full CRUD operations (create, retrieve, cancel, list)
- Implement background batch processing with configurable concurrency
- Add SQLite KVStore backend for persistence
- Support /v1/chat/completions endpoint with request validation

Comprehensive Test Suite:
- Add unit tests for provider implementation with validation
- Add integration tests for end-to-end batch processing workflows
- Add error handling tests for validation, malformed inputs, and edge cases

Configuration:
- Add max_concurrent_batches and max_concurrent_requests_per_batch options
- Add provider documentation with sample configurations

Test with -

```
$ uv run llama stack build --image-type venv --providers inference=YOU_PICK,files=inline::localfs,batches=inline::reference --run &
$ LLAMA_STACK_CONFIG=http://localhost:8321 uv run pytest tests/unit/providers/batches tests/integration/batches --text-model YOU_PICK
```

addresses #3066

### CODE REVIEW

**Comment by ashwinb** [comment]
undid the recordings, now using the new recording workflow, let's see

**Comment by ashwinb** [comment]
This run - https://github.com/llamastack/llama-stack/actions/runs/17000107360/job/48200004734 - worked and uploaded. Note I had to invoke it a bit subtly:

- Go to Actions, find the Record workflow
- This PR is in the upstream repository, but if you had a public fork you need to go to Actions in your fork!
- Choose Run Workflow
- Then choose this PR's branch as the branch to run the workflow from
- Submit

This is all too tricky for most people. Writing a script for this now. I am going through all this trouble because I think remote recordings are very important. And Fast CI is very important to me.

File: tests/integration/batches/test_batches.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (ashwinb) [inline tests/integration/batches/test_batches.py:1]: These tests don't work with "library client" due to two reasons

- instead of `openai_client`, the fixture should have been `compat_client` (of course, undocumented gah)
- even if that fix were made, library client will use the `llama-stack-client-python` SDK which hasn't been updated yet with the new client-side types

I am going to somehow figure out a way to exclude this directory from the library-client tests for now.

**Review by ashwinb — APPROVED** [review]

**Comment by ashwinb** [comment]
all green!
